cypress/e2e/GUI/gui.cy.js

```diff
@@ -0,0 +1,230 @@
+        cy.get('.ThemeToggle_container__RIfhe').click()
+        cy.get('body[data-theme="light"]').should('be.visible')
+    })
+
+    it('Verifica se o tema se mantem ao acessar a informacao de cliente corretamente', () => {
+
+        //Clica no bot�o de trocar o tema e checa se trocou corretamente para Dark
+        cy.get('.ThemeToggle_container__RIfhe').click()
+        cy.get('body[data-theme="dark"]').should('be.visible')
+
+        //Clica para acessar os detalhes de um cliente e verifica se foi realizado corretamente
+        cy.contains('button', 'View').click()
+        cy.contains('Customer Details').should('be.visible')
+        cy.get('body[data-theme="dark"]').should('be.visible')
+    })
+
+    it('Mantem os filtros ao voltar da visualizacao de detalhes do cliente', () => {
+
+        //Realiza os filtros corretamente
+        cy.get('select[data-testid="size-filter"]').select('Small')
+        cy.get('select[data-testid="industry-filter"]').select('Logistics')
+
+        //Acessa um cliente apresentado e checa se acessou
+        cy.contains('button', 'View').click()
+        cy.contains('Customer Details').should('be.visible')
+
+        //Retorna e confirma que retornou para a p�gina anterior
+        cy.contains('button', 'Back').click({ force: true })
+        cy.contains('Below is our customer list.').should('be.visible')
+
+        //Verifica se o filtro segue os mesmos selecionados anteriormente
+        cy.get('select[data-testid="size-filter"]').should('have.value', 'Small')
+        cy.get('select[data-testid="industry-filter"]').should('have.value', 'Logistics')
+    })
+
+    it('Verifica se e possivel acessar os detalhes do cliente', () => {
+
+        //Clica para acessar os detalhes de um cliente e verifica se foi realizado corretamente
+        cy.contains('button', 'View').click()
+        cy.contains('Customer Details').should('be.visible')
+    })
+
+    it('Verifica se e possivel expandir e recolher o endereco do cliente', () => {
+
+        //Clica para acessar os detalhes de um cliente e verifica se foi realizado corretamente
+        cy.contains('button', 'View').click()
+        cy.contains('Customer Details').should('be.visible')
+
+        //Clica no bot�o "Show Address" para expandir o endere�o do cliente e verifica se expandiu
+        cy.contains('Show address').click({ force: true })
+        cy.contains('Hide address').should('be.visible')
+
+        //Clica no bot�o "Hide Address" para recolher o endere�o do cliente e verifica se recolheu
+        cy.contains('Hide address').click({ force: true })
+        cy.contains('Show address').should('be.visible')
+    })
+
+    it('Retorna a lista de clientes ao clicar no botao "Voltar"', () => {
+        //Verifica se as informa��es do rodap� est�o sendo apresentadas corretamente
+        cy.contains('button', 'View').click()
+        cy.contains('Customer Details').should('be.visible')
+        cy.contains('button', 'Back').click({ force: true })
```

Comment:
Aqui também.

`force: true` força o clique em um elemento que a princípio não seria possível de ser clicado por um usuário real, e testes com Cypress devem simular o uso real da aplicação.